### PR TITLE
Added new gemspec parser.

### PR DESCRIPTION
**Pull request by ankk13**
Parser reads the file ending with .gemspec. 
Refers value in it or the filepath to get name and version of the package.

### CODE REVIEW

File: pkg/gemspec/parse_testcase.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+	}
+	JsonGem = []types.Library{
+		{"json", "2.3.0", "Ruby"},
+	}
+	JsonVersionGem = []types.Library{
+		{"json", "VERSION", "Ruby"},
+	}
+	JsonNameGem = []types.Library{
+		{"$json", "2.3.0", "Ruby"},
+	}
```
Comment (knqyf263) [inline pkg/gemspec/parse_testcase.go:17]: You can merge them into parse_test.go.

File: pkg/gemspec/testdata/json-2.3.0.gemspec [hunk 2]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env jruby
+require "rubygems"
+
+spec = Gem::Specification.new do |s|
+  s.name = "$json"
```
Comment (knqyf263) [inline pkg/gemspec/testdata/json-2.3.0.gemspec:5]: Could you tell me the spec of the name field? Any documentation? I want to know why `$` is accepted here.

File: pkg/gemspec/parse.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	"unicode/utf8"
+
+	"github.com/aquasecurity/go-dep-parser/pkg/types"
+	"golang.org/x/xerrors"
+)
+
+var (
+	bwQuotes         = regexp.MustCompile(`['|"](.*?)['|"]`)
+	gemSpec          = ".gemspec"
+	gemStr           = ".name = "
```
Comment (knqyf263) [inline pkg/gemspec/parse.go:18]: Spaces are mandatory? I wonder if `.name="foo"` is allowed.

File: pkg/gemspec/parse.go [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+	gemStrVer        = ".version = "
+	gemStrLic        = ".licenses = "
+	packageNameRegEx = regexp.MustCompile(`^[a-zA-Z0-9_\-]+$`)
+	versionRegEx     = regexp.MustCompile(`^[0-9.]+$`)
+	fileNameVersion  = regexp.MustCompile(`^([\w_.\-]+)-([0-9.]+)$`)
+)
+
+func Parse(r io.Reader, filePath string) ([]types.Library, error) {
```
Comment (knqyf263) [inline pkg/gemspec/parse.go:26]: If it always returns a single library, you don't have to use []types.Library.

```suggestion
func Parse(r io.Reader, filePath string) (types.Library, error) {
```

File: pkg/gemspec/testdata/json-2.3.0.gemspec (hunk 2 above)
Comment (ankk13) [reply pkg/gemspec/testdata/json-2.3.0.gemspec:5]: you can refer https://guides.rubygems.org/specification-reference/#name. It suggests formal syntax.
To explore, I tried below commented regexes on image kankush/ruby:v2. This image has famous ruby packages installed.
I did not miss the name from any gemspec file. But I still think regexes should be used, as this is not strict format. 

$json is no different than simple package name. So should be considered.

File: pkg/gemspec/parse.go (hunk 3 above)
Comment (ankk13) [reply pkg/gemspec/parse.go:18]: addressed in above comment

**Comment by knqyf263** [comment]
Close this PR in favor of https://github.com/aquasecurity/go-dep-parser/pull/17

**Pull request closed without merging**